pydy/codegen/ode_function_generators.py

```diff
@@ -885,11 +898,11 @@ def __init__(self, *args, **kwargs):
 
     def define_inputs(self):
 
-        if self.specifieds is None:
-            self.inputs = self.coordinates + self.speeds + self.constants
-        else:
-            self.inputs = (self.coordinates + self.speeds + self.specifieds
-                           + self.constants)
+        specifieds = []
+        if self.specifieds is not None:
+            specifieds = self.specifieds
+        self.inputs = chain(self.coordinates, self.speeds, specifieds,
```

Comment:
I don't know. It looks like that both methods do the same thing. Maybe the one in the Theano generator is a remnant that I forgot to remove. It looks like you can delete it.